onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -3732,16 +3732,40 @@ def aten_logaddexp2(self: TFloatOrBFloat16, other: TFloatOrBFloat16) -> TFloatOr
     return op.Div(op.Log(summation), op.Log(2.0))
 
 
-@torch_op("aten::logcumsumexp")
-def aten_logcumsumexp(self: TFloatOrBFloat16, dim: INT64) -> TFloatOrBFloat16:
+@torch_op("aten::logcumsumexp", trace_only=True)
+def aten_logcumsumexp(self: TFloatOrBFloat16, dim: int) -> TFloatOrBFloat16:
     """logcumsumexp(Tensor self, int dim) -> Tensor"""
 
-    if op.Size(op.Shape(self)) == 0:
+    return _aten_logcumsumexp_onnx(self, [dim])
+
+@torch_op("aten::logcumsumexp", private=True)
+def _aten_logcumsumexp_onnx(self: TFloatOrBFloat16, dims: Sequence[int]) -> TFloatOrBFloat16:
+    """logcumsumexp(Tensor self, int dim) -> Tensor
+
+    Iterative implementation based on https://www.tensorflow.org/api_docs/python/tf/math/cumulative_logsumexp
+    for stability.
+    """
+
+    @graph()
+    def log_add_exp(x, y):
+        # Compute pairwise log_exp.
+        # Based on https://www.tensorflow.org/api_docs/python/tf/math/cumulative_logsumexp.
+        # log_add_exp(x, y) = log(1 + exp(min(x, y) - max(x, y))) + max(x, y)
+        sum_out = op.Add(op.Log(op.Add(1, op.Exp(op.Sub(op.Min(x, y), op.Max(x, y))))), op.Max(x, y))
+        return x, sum_out
```

Comment:
This doesn't sound right. It should probably be `return sum_out, sum_out` ?

Reply:
The signature of the function is supposed to be
```
   Accumulated-Value-In X Next-Input => Accumulated-Value-Out X Next-Output
```
It allows for the returned output to be different from the accumulated-values, but for simple-reduction we will have both be the same